src/fu-polkit-agent.c

```diff
@@ -0,0 +1,219 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2011 Lennart Poettering <lennart@poettering.net>
+ * Copyright (C) 2012 Matthias Klumpp <matthias@tenstral.net>
+ * Copyright (C) 2015-2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include <stdio.h>
+#include <sys/types.h>
+#ifdef __linux__
+#include <sys/prctl.h>
+#endif
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <dirent.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <sys/poll.h>
+#include <sys/wait.h>
+#include <glib.h>
+
+#include "fu-polkit-agent.h"
+
+#define POLKIT_TTY_AGENT_BINARY_PATH "/usr/bin/pkttyagent"
```

Comment:
Probably shouldn't hardcode this path, we have infrastructure to lookup paths as configured by distros.

Reply:
Fixed.